src/west/app/project.py

```diff
@@ -1363,13 +1373,40 @@ def update_submodules(self, project):
                              '--', submodule.path])
             ref = []
             if (cache_dir):
+                # check if the submodule cache is present within the project cache
                 submodule_ref = Path(cache_dir, submodule.path)
-                if any(os.scandir(submodule_ref)):
+                if submodule_ref.is_dir() and any(os.scandir(submodule_ref)):
                     ref = ['--reference', os.fspath(submodule_ref)]
                     self.small_banner(f'using reference from: {submodule_ref}')
                     self.dbg(
                         f'found {submodule.path} in --path-cache {submodule_ref}',
                         level=Verbosity.DBG_MORE)
+                else:
+                    # determine the url from the submodule and handle it like
+                    # any other project cache
+                    res = project.git(['config', '--file', '.gitmodules',
+                                       f'submodule.{submodule.path}.url'],
+                                      capture_stdout=True)
+                    if not res.stdout or res.returncode:
+                        self.die("Submodule url cannot be determined for project:"
+                                 f"{project.name} ({project.path}).")
+                    submodule_url = res.stdout.decode('utf-8').strip()
+
+                    submodule_project = Project(name=submodule.name,
+                                                path=submodule.path,
+                                                url=submodule_url)
```

Comment:
Not a huge fan of this artificially created project to have submodules cached next to actual projects.
Instead, can the project bare clones keep submodules in sync?